src/request/sign-transaction/BaseLayout.js

```diff
@@ -11,51 +11,69 @@ class BaseLayout {
      * @param {KeyguardRequest.ParsedSignTransactionRequest} request
      * @param {Function} resolve
      * @param {Function} reject
+     * @param {HTMLElement} $recipient
      */
-    constructor(request, resolve, reject) {
-        /** @type {HTMLDivElement} */
-        const $pageBody = (document.querySelector('#confirm-transaction .transaction'));
+    constructor(request, resolve, reject, $recipient) {
+        /** @type {HTMLElement} */
+        this.$el = (document.getElementById('layout-container'));
+        this.$el.classList.add(request.layout);
 
-        /** @type {HTMLDivElement} */
-        const $senderIdenticon = ($pageBody.querySelector('#sender-identicon'));
-        /** @type {HTMLDivElement} */
-        const $recipientIdenticon = ($pageBody.querySelector('#recipient-identicon'));
+        /** @type {HTMLElement} */
+        const $recipientNode = (this.$el.querySelector('.account.recipient'));
+        /** @type {HTMLElement} */
+        ($recipientNode.parentElement).replaceChild($recipient, $recipientNode);
 
-        /** @type {HTMLDivElement} */
-        const $senderLabel = ($pageBody.querySelector('#sender-label'));
-        /** @type {HTMLDivElement} */
-        const $recipientLabel = ($pageBody.querySelector('#recipient-label'));
+        const transaction = request.transaction;
 
+        // sender
         /** @type {HTMLDivElement} */
-        const $senderAddress = ($pageBody.querySelector('#sender-address'));
-        /** @type {HTMLDivElement} */
-        const $recipientAddress = ($pageBody.querySelector('#recipient-address'));
+        const $sender = (this.$el.querySelector('.sender'));
 
         /** @type {HTMLDivElement} */
-        const $value = ($pageBody.querySelector('#value'));
-        /** @type {HTMLDivElement} */
-        const $fee = ($pageBody.querySelector('#fee'));
-        /** @type {HTMLDivElement} */
-        const $data = ($pageBody.querySelector('#data'));
+        const $senderIdenticon = ($sender.querySelector('.identicon'));
+        // eslint-disable-next-line no-new
+        new Identicon(transaction.sender.toUserFriendlyAddress(), $senderIdenticon);
+
+        const $senderAddresses = ($sender.querySelectorAll('.address > .chunk'));
+        /** @type {string[]} */
+        const senderAddressChunks = (
+            transaction.sender
+                .toUserFriendlyAddress()
+                .replace(/[+ ]/g, '').match(/.{4}/g)
+        );
+        $senderAddresses.forEach(($el, x) => {
+            $el.textContent = senderAddressChunks[x];
+        });
```

Comment:
I feel like the chunked address could be a standalone component that encapsulates the splitting and display and copying. Maybe in another PR.